src/query/expression/src/type_check.rs

```diff
@@ -745,3 +744,72 @@ pub const ALL_SIMPLE_CAST_FUNCTIONS: &[&str] = &[
 pub fn is_simple_cast_function(name: &str) -> bool {
     ALL_SIMPLE_CAST_FUNCTIONS.contains(&name)
 }
+
+/// notes:
+/// - this is NOT can_auto_cast_to.
+/// - this should be consistent with run_cast, partially ensured by test: test_can_cast_to .
+pub fn can_cast_to(src_type: &DataType, dest_type: &DataType) -> bool {
+    use DataType::*;
+    if src_type == dest_type {
+        return true;
+    }
+    // we mainly care about which types can/cannot cast to dest_type.
+    // the match is written in a way to make it easier to read this info.
+    // try to use less _ to avoid miss something
+    match (dest_type, src_type) {
```

Comment:
seems put `src_type` on left is more customary.

Reply:
the comments explain the reason.

```
// we mainly care about which types can/cannot cast to dest_type.
// the match is written in a way to make it easier to read this info.
```

Reply:
To be more specific, our use case is as follows: we have a dest type and several src types. We need to determine if the cast between them is valid.